h2o-algos/src/main/java/hex/tree/isoforextended/ExtendedIsolationForest.java

```diff
@@ -171,9 +179,22 @@ private void buildIsolationTreeEnsemble() {
                 DKV.put(compressedIsolationTree);
                 _job.update(1);
                 _model.update(_job);
-                LOG.info((tid + 1) + ". tree was built in " + timer.toString());
+                _model._output._training_time_ms[tid + 1] = System.currentTimeMillis();
+                LOG.info((tid + 1) + ". tree was built in " + timer);
                 isolationTreeStats.updateBy(isolationTree);
+
+                boolean manualInterval = _parms._score_tree_interval > 0 && (tid +1) % _parms._score_tree_interval == 0;
+                boolean finalScoring = _parms._ntrees == (tid + 1);
+
+                _model._output._scored_train[tid + 1] = new ScoreKeeper();
+                if (_parms._score_each_iteration || manualInterval || finalScoring) {
```

Comment:
Is it correct that if this condition is false, then the scoreKeeper is not set and is empty?

Reply:
Yes, It should be more clever here, I will reconsider the approach, thank you!

Reply:
It is fixed in following PR here https://github.com/h2oai/h2o-3/pull/6039. Actually, it is the way how SharedTree is creating scoring history now in SharedTreeModel:242. It is leaving empty ScoreKeeper 🤔